### PR TITLE
Implement PlotTime

### PR DESCRIPTION
**Pull request by WinPooh32**
Implemented PlotTime as extra type.
extra method ` (i PlotTime) Time()` returns Go's native `time.Time`

It's required for these widgets:

- TimePicker
- DatePicker

### CODE REVIEW

File: extra_types.go [hunk 1]
```diff
@@ -176,6 +179,32 @@ func (p PlotPoint) toC() C.ImPlotPoint {
 	return C.ImPlotPoint{x: C.double(p.X), y: C.double(p.Y)}
 }
 
+type PlotTime struct {
+	S  int // second part
+	Us int // microsecond part
+}
```
Comment (gucio321) [inline extra_types.go:185]: what about making it just `type PlotTime time.Time`? It'll enable type-casting so from `time.Time` to `PlotTime` and the usage'd be easier imao

**Review by gucio321 — COMMENTED** [review]
Hi!
thank you for PR!
it looks good, just one suggestion below

File: extra_types.go (hunk 1 above)
Comment (WinPooh32) [reply extra_types.go:185]: PlotTime has no time zone offset, but `time.Time` has.
PlotTime has lower precision (microseconds vs nanoseconds at `time.Time`) 
It all can cause misunderstand how to use it.
Also no need to make arithmetic operations on every C call at current implementation.

Comment (gucio321) [reply extra_types.go:185]: ok, sounds fair.

**Review by gucio321 — APPROVED** [review]

**Pull request merged**